Assets/MRTK/SDK/Features/UX/Scripts/RiggedHandVisualizer/RiggedHandVisualizer.cs

```diff
@@ -301,178 +287,143 @@ private void OnDisable()
+                    float ringFingerCurl = HandPoseUtils.RingFingerCurl(Controller.ControllerHandedness);
+                    float pinkyFingerCurl = HandPoseUtils.PinkyFingerCurl(Controller.ControllerHandedness);
 
-                if (handMaterial != null && handRendererInitialized)
-                {
-                    float gripStrength = indexFingerCurl + middleFingerCurl + ringFingerCurl + pinkyFingerCurl;
-                    gripStrength /= 4.0f;
-                    gripStrength = gripStrength > 0.8f ? 1.0f : gripStrength;
+                    if (handMaterial != null && handRendererInitialized)
+                    {
+                        float gripStrength = indexFingerCurl + middleFingerCurl + ringFingerCurl + pinkyFingerCurl;
+                        gripStrength /= 4.0f;
+                        gripStrength = gripStrength > 0.8f ? 1.0f : gripStrength;
 
-                    pinchStrength = Mathf.Pow(Mathf.Max(pinchStrength, gripStrength), 2.0f);
+                        pinchStrength = Mathf.Pow(Mathf.Max(pinchStrength, gripStrength), 2.0f);
 
-                    if (handRenderer.sharedMaterial.HasProperty(pinchStrengthMaterialProperty))
-                    {
-                        handRenderer.sharedMaterial.SetFloat(pinchStrengthMaterialProperty, pinchStrength);
-                    }
-                    // Only show this warning once
-                    else if (!displayedMaterialPropertyWarning)
-                    {
-                        Debug.LogWarning(String.Format("The property {0} for reacting to pinch strength was not found. A material with this property is required to visualize pinch strength.", pinchStrengthMaterialProperty));
-                        displayedMaterialPropertyWarning = true;
+                        if (handRenderer.sharedMaterial.HasProperty(pinchStrengthMaterialProperty))
+                        {
+                            handRenderer.sharedMaterial.SetFloat(pinchStrengthMaterialProperty, pinchStrength);
+                        }
+                        // Only show this warning once
+                        else if (!displayedMaterialPropertyWarning)
+                        {
+                            Debug.LogWarning(String.Format("The property {0} for reacting to pinch strength was not found. A material with this property is required to visualize pinch strength.", pinchStrengthMaterialProperty));
+                            displayedMaterialPropertyWarning = true;
+                        }
                     }
                 }
             }
         }
 
+        /// <inheritdoc/>
+        public override void OnHandMeshUpdated(InputEventData<HandMeshInfo> eventData)
+        {
+            // We've gotten hand mesh data from the platform, don't use the rigged hand models for visualization
+            if (eventData.Handedness != Controller?.ControllerHandedness)
```

Comment:
Should this be checking for equality? 
```suggestion
            if (eventData.Handedness == Controller?.ControllerHandedness)
```